modules/widgets/blogs-i-follow/blogs-i-follow.php

```diff
@@ -0,0 +1,540 @@
+		}
+
+		return $output;
+	}
+
+	/**
+	 * output for friendly message
+	 * displayed when there are no subscriptions and
+	 * the current user is the user to whom the widget belongs
+	 *
+	 * @return string the message
+	 */
+	function get_friendly_message() {
+		$message = sprintf(
+			__( 'You are not yet following any blogs. Try <a href="%1$s">finding your friends</a> or check out our <a href="%2$s">recommended blogs</a>.', 'jetpack' ),
+			esc_url( apply_filters('jetpack_blog_locale', null) . '.wordpress.com/find-friends' ) . '" target="_blank',
+			esc_url( apply_filters('jetpack_blog_locale', null) . '.wordpress.com/recommendations' ) . '" target="_blank'
+		);
+
+		return '<p>' . $message . '</p>';
+	}
+
+	/**
+	 * widget save update
+	 *
+	 * @param array $new_instance the widget instance being saved
+	 * @param array $old_instance the widget instance, prior to being saved
+	 * @return array $instance, the saved widget instance
+	 */
+	function update( $new_instance, $old_instance ) {
+		$instance = array();
+
+		$instance['title']  = wp_kses( $new_instance['title'], array() );
+		$instance['number'] = absint( $new_instance['number'] );
+
+		$instance['user_id'] = (int) $new_instance['user_id'];
+		if ( empty( $instance['user_id'] ) ) {
+			$instance['user_id'] = absint( get_current_user_id() );			
+		}
+
+		if ( $instance['number'] < 1 || $instance['number'] > 50 ) {
+			$instance['number'] = 20;
+		}
+
+		$instance['display'] = isset( $new_instance['display'] ) && 'grid' == $new_instance['display'] ? 'grid' : 'list';
+
+		// void the transients
+		delete_transient( $this->id . '-subscriptions' );
+		delete_transient( $this->id . '-widget' );
+
+		// generate the first set of subscriptions
+		$this->get_subscriptions();
+
+		return $instance;
+	}
+
+	/**
+	 * display the widget admin form
+	 *
+	 * @param array $instance the current widget instance
+	 *
+	 * @return void
+	 */
+	function form( $instance ) {
+		$instance = $this->extend_default_options( $instance );
+		$display = $instance['display'];
+
+		if ( empty( $instance['user_id']  ) ) {
+			$instance['user_id'] = get_current_user_id();
+		}
+		?>
+		<p>
+			<label for="<?php echo $this->get_field_id( 'title' ); ?>"><?php esc_html_e( 'Title:', 'jetpack' ); ?></label>
+			<input type="text" class="widefat" id="<?php echo $this->get_field_id( 'title' ); ?>" name="<?php echo $this->get_field_name( 'title' ); ?>" value="<?php echo esc_attr( $instance['title'] ); ?>" />
+		</p>
+
+		<p>
+			<label for="<?php echo $this->get_field_id( 'number' ); ?>"><?php esc_html_e( 'Number of blogs to show:', 'jetpack' ); ?>
+				<input type="number" id="<?php echo $this->get_field_id( 'number' ); ?>" name="<?php echo $this->get_field_name( 'number' ); ?>" value="<?php echo esc_attr( $instance['number'] ); ?>" min="1" max="50" />
+				<br><small><?php esc_html_e( '(at most 50)', 'jetpack' ) ?> <a href="https://en.support.wordpress.com/widgets/blogs-i-follow-widget/" target="_blank">( ? )</a></small>
```

Comment:
I think there might be a better way to translate this.  (at most 50) ( ? ).
Some languages do not use spaces to differentiate between words. So things like spaces and ? should be part of the string that gets translated.